### PR TITLE
Feat: Add Support for OCI

### PR DESCRIPTION
**Pull request by theBeginner86**
**Description**

Adds functionality to:
1. Compress into OCI Artifacts
2. De-Compress OCI Artifacts

This would be helpful for importing/exporting Designs and Models.

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/oci/oci.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		o.meta.Created = ct.Format(time.RFC3339)
+	}
+
+	img := mutate.MediaType(empty.Image, types.OCIManifestSchema1)
+	img = mutate.ConfigMediaType(img, oci.CanonicalConfigMediaType)
+	img = mutate.Annotations(img, o.meta.ToAnnotations()).(gcrv1.Image)
+
+	img, err = mutate.Append(img, mutate.Addendum{Layer: layer})
+	if err != nil {
+		return nil, fmt.Errorf("appeding content to artifact failed: %w", err)
```
Comment (leecalcote) [inline models/oci/oci.go:106]:
```suggestion
		return nil, fmt.Errorf("appending content to artifact failed: %w", err)
```

File: models/oci/build.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/layer5io/meshkit/models/oci/fs"
+	"github.com/fluxcd/pkg/sourceignore"
+)
+
+
+func build(artifactPath, sourceDir string, ignorePaths []string) (err error) {
```
Comment (leecalcote) [inline models/oci/build.go:18]:
```suggestion
// Supports local folder (directory) path only. 
func build(artifactPath, sourceDir string, ignorePaths []string) (err error) {
```

**Comment by leecalcote** [comment]
@MUzairS15 you're going to want to review this.

**Review by MUzairS15 — COMMENTED** [review]
`errors`  needs to be transformed into `Meshkit errors`.
I wonder if it's better to directly import theire packages or else if we are directly sourcing form their repo, then appropriate licensing needs to be added in the code.
// @theBeginner86

**Comment by theBeginner86** [comment]
> `errors` needs to be transformed into `Meshkit errors`. I wonder if it's better to directly import theire packages or else if we are directly sourcing form their repo, then appropriate licensing needs to be added in the code. // @theBeginner86

Added MeshKit errors.
Removed duplicated code and imported the necessary functions.

**Comment by leecalcote** [comment]
Re-running build...

**Comment by leecalcote** [comment]
This is ready, right, @theBeginner86?

**Comment by leecalcote** [comment]
I tried, but...

**Comment by theBeginner86** [comment]
> Build failure...

Fixed: https://github.com/meshery/meshkit/actions/runs/7462230856/job/20304075337?pr=443

**Pull request merged**